### PR TITLE
Upsert support for dbs

### LINKED ISSUE #1768: Upsert only works for PostgreSQL
Related to PR  https://github.com/hibernate/hibernate-reactive/pull/1765

We added an upsert method to the stateless session interfaces in Hibernate Reactive.
For example in [Mutiny.StatelessSession#upsert](https://github.com/hibernate/hibernate-reactive/blob/1b0ec2be834bcdeadd8106dc560514e10b45548d/hibernate-reactive-core/src/main/java/org/hibernate/reactive/mutiny/Mutiny.java#L1747).

The problem is that it seems to only work for PostgreSQL.
The [test](https://github.com/hibernate/hibernate-reactive/blob/1b0ec2be834bcdeadd8106dc560514e10b45548d/hibernate-reactive-core/src/test/java/org/hibernate/reactive/UpsertTest.java#L39) fails for the other databases because when it looks for the results in the db, it returns an empty list.

I've quickly checked the same test with Hibernate ORM and MySQL, and it seems to run different queries. Probably, not all databases support the operation.  But we need to look into it.

### PR DESCRIPTION
**Pull request by blafond**
fixes #1768

### CODE REVIEW

File: hibernate-reactive-core/src/main/java/org/hibernate/reactive/sql/model/ReactiveOptionalTableUpdateOperation.java [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+	/**
+	 * @see org.hibernate.sql.model.jdbc.OptionalTableUpdateOperation#performUpdate(JdbcValueBindings, SharedSessionContractImplementor)
+	 * @param tableMapping
+	 * @param jdbcValueBindings
+	 * @param valuesAnalysis
+	 * @param session
+	 * @return
+	 */
+	private CompletionStage<Boolean> performReactiveUpdate(
+			TableMapping tableMapping,
```
[the remark this answers is not in the document]
Comment (blafond) [reply hibernate-reactive-core/src/main/java/org/hibernate/reactive/sql/model/ReactiveOptionalTableUpdateOperation.java]: fixed

File: hibernate-reactive-core/src/main/java/org/hibernate/reactive/sql/model/ReactiveDeleteOrUpsertOperation.java [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+
+		ReactiveConnection reactiveConnection = ( (ReactiveConnectionSupplier) session ).getReactiveConnection();
+		String sqlString = statementDetails.getSqlString();
+		return reactiveConnection
+				.update( sqlString, params ).thenCompose(this::checkUpsertResults);
+	}
+
+	private void bindDeleteKeyValues(
+			JdbcValueBindings jdbcValueBindings,
+			List<ColumnValueParameter> parameters,
```
[the remark this answers is not in the document]
Comment (blafond) [reply hibernate-reactive-core/src/main/java/org/hibernate/reactive/sql/model/ReactiveDeleteOrUpsertOperation.java]: fixed

File: hibernate-reactive-core/src/main/java/org/hibernate/reactive/sql/model/ReactiveOptionalTableUpdateOperation.java [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+			JdbcValueBindings jdbcValueBindings,
+			UpdateValuesAnalysis valuesAnalysis,
```
[the remark this answers is not in the document]
Comment (blafond) [reply hibernate-reactive-core/src/main/java/org/hibernate/reactive/sql/model/ReactiveOptionalTableUpdateOperation.java]: fixed

**Comment by DavideD** [comment]
Superseeded by https://github.com/hibernate/hibernate-reactive/pull/1805

**Pull request closed without merging**